Mathlib/RingTheory/HopfAlgebra/GroupLike.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Michał Mrugała. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Michał Mrugała
+-/
+import Mathlib.RingTheory.HopfAlgebra.Basic
+import Mathlib.RingTheory.Bialgebra.GroupLike
+
+/-!
+# Group-like elements in a Hopf algebra
+
+This file proves that group-like elements in a Hopf algebra form a group.
+-/
+
+open HopfAlgebra
+
+variable {R A : Type*}
+
+section Semiring
+variable [CommSemiring R] [Semiring A] [HopfAlgebra R A] {a b : A}
+
+@[simp] lemma IsGroupLikeElem.antipode_mul_cancel (ha : IsGroupLikeElem R a) :
+    antipode R a * a = 1 := by
+  simpa [ha, -mul_antipode_lTensor_comul_apply] using mul_antipode_rTensor_comul_apply (R := R) a
+
+@[simp] lemma IsGroupLikeElem.mul_antipode_cancel (ha : IsGroupLikeElem R a) :
+    a * antipode R a = 1 := by
+  simpa [ha, -mul_antipode_lTensor_comul_apply] using mul_antipode_lTensor_comul_apply (R := R) a
+
+variable (R) in
+/-- Turn a group-like element `a` into a unit with inverse its antipode. -/
+@[simps]
+def GroupLike.toUnits : GroupLike R A →* Aˣ where
+  toFun a := {
+    val := a
+    inv := antipode R a
+    val_inv := a.2.mul_antipode_cancel
+    inv_val := a.2.antipode_mul_cancel
+  }
```

Comment:
```suggestion
  toFun a :=
  { val := a
    inv := antipode R a
    val_inv := a.2.mul_antipode_cancel
    inv_val := a.2.antipode_mul_cancel }
```

Reply:
I do not like this style because it makes it harder to edit fields